map/generate_chart.cpp

```diff
@@ -0,0 +1,127 @@
+#include "map/generate_chart.hpp"
+
+//#include "coding/png_memory_encoder.hpp"
+
+#include "base/assert.hpp"
+
+#include "std/fstream.hpp"
+
+#include "3party/agg/agg_conv_stroke.h"
+#include "3party/agg/agg_path_storage.h"
+#include "3party/agg/agg_pixfmt_rgba.h"
+#include "3party/agg/agg_rasterizer_scanline_aa.h"
+#include "3party/agg/agg_renderer_primitives.h"
+#include "3party/agg/agg_renderer_scanline.h"
+#include "3party/agg/agg_scanline_p.h"
+
+namespace
+{
+template <class TColor, class TOrder>
+struct TBlendAdaptor
+{
+  using order_type = TOrder;
```

Comment:
I can not. BlendAdaptor is used in other agg template:
agg::pixfmt_custom_blend_rgba<BlendAdaptor<agg::rgba8, agg::order_rgba>, agg::rendering_buffer>
and agg::pixfmt_custom_blend_rgba wants BlendAdaptor to have types with names order_type and color_type